### ISSUE TITLE
support http.extraHeader in .git/config

### ISSUE THREAD

**Issue by wmhilton** [isomorphic-git/isomorphic-git#530]
https://git-scm.com/docs/git-config#git-config-httpextraHeader

Requested by @juancampa [here](https://github.com/isomorphic-git/isomorphic-git/issues/500#issuecomment-429878520)

Will pair nicely with #502

**Comment by juancampa** [comment]
Sweet! There are two version of this option. One general for all remotes (`http.extraHeader`) but also remote-specific (`http.<remote>.extraHeader`). At the moment I only need the general one since I only push to one remote.

Thanks @wmhilton!

**Comment by wmhilton** [comment]
Yes... and it _might_ make sense to allow specifying it as a function argument to `fetch`/`push`/`clone` too. Having to write it to the config first seems awkward. On the other hand, I don't want to stray too far from the canonical implementation.

**Comment by juancampa** [comment]
Yeah, that too, I actually use `git -c http.extraHeader="..."` as opposed to the config

**Comment by juancampa** [comment]
Hey @wmhilton, do you mind if I work on this? I'm at a point where this is blocking me so if you haven't started, I wouldn't mind giving it a shot.

**Comment by juancampa** [comment]
> Yes... and it _might_ make sense to allow specifying it as a function argument to `fetch`/`push`/`clone` too. Having to write it to the config first seems awkward. On the other hand, I don't want to stray too far from the canonical implementation.

This makes me think that maybe we need the equivalent of git's `-c` as a generic way to pass ad-hoc config values to any command
